! I have used "NewsApi "  for data harvesting for Tesla news



In [9]:
from newsapi import NewsApiClient
import pandas as pd
import json
import requests

In [23]:
url='https://newsapi.org/v2/everything?q=tesla&from=2021-05-18&sortBy=publishedAt&apiKey=7786d52ad23347b4bcb23b3545b3d268'
data = requests.get(url).json()
print(data)
type(data)

{'status': 'ok', 'totalResults': 16938, 'articles': [{'source': {'id': 'nrk', 'name': 'NRK'}, 'author': 'Lars Håkon Pedersen', 'title': 'Kinesisk sjarmoffensiv mot norske elbilkunder', 'description': 'Elbilmarkedet i Norge har vært dominert av tyske, amerikanske, franske, japanske og koreanske biler. Nå legger kineserne bredsiden til for å vinne norske elbilhjerter.', 'url': 'https://www.nrk.no/norge/na-kommer-de-kinesiske-elbilene-for-alvor-1.15535790', 'urlToImage': 'https://gfx.nrk.no/cL1W3jptC9CW0XYwWVlrSgy4zzmodPTI5OtsbBHdtv1w.jpg', 'publishedAt': '2021-06-18T21:43:59Z', 'content': 'BYD, NIO, Hongqi, Xpeng, Seres og Dongfeng. Kanskje ikke så kjente bilmerker for nordmenn flest, men snart kan de bli det.\r\nFor helkinesiske elbiler er på full fart inn i det norske markedet, bekreft… [+5142 chars]'}, {'source': {'id': None, 'name': 'Mdig.com.br'}, 'author': 'https://www.facebook.com/luisaocs', 'title': "A antiga roda d'água revisitada como uma bomba", 'description': 'A eletricidade 

dict

#Here we have total result of 16938 with data type dictonary

In [50]:
#Just checking keys and article1
print(data.keys())
print(data['articles'][0])

dict_keys(['status', 'totalResults', 'articles'])
{'source': {'id': 'nrk', 'name': 'NRK'}, 'author': 'Lars Håkon Pedersen', 'title': 'Kinesisk sjarmoffensiv mot norske elbilkunder', 'description': 'Elbilmarkedet i Norge har vært dominert av tyske, amerikanske, franske, japanske og koreanske biler. Nå legger kineserne bredsiden til for å vinne norske elbilhjerter.', 'url': 'https://www.nrk.no/norge/na-kommer-de-kinesiske-elbilene-for-alvor-1.15535790', 'urlToImage': 'https://gfx.nrk.no/cL1W3jptC9CW0XYwWVlrSgy4zzmodPTI5OtsbBHdtv1w.jpg', 'publishedAt': '2021-06-18T21:43:59Z', 'content': 'BYD, NIO, Hongqi, Xpeng, Seres og Dongfeng. Kanskje ikke så kjente bilmerker for nordmenn flest, men snart kan de bli det.\r\nFor helkinesiske elbiler er på full fart inn i det norske markedet, bekreft… [+5142 chars]'}


In [62]:
#key value for article1 
for key , value in articles[0].items():
    print(f"\n{key.ljust(15)} {value}")


source          {'id': 'nrk', 'name': 'NRK'}

author          Lars Håkon Pedersen

title           Kinesisk sjarmoffensiv mot norske elbilkunder

description     Elbilmarkedet i Norge har vært dominert av tyske, amerikanske, franske, japanske og koreanske biler. Nå legger kineserne bredsiden til for å vinne norske elbilhjerter.

url             https://www.nrk.no/norge/na-kommer-de-kinesiske-elbilene-for-alvor-1.15535790

urlToImage      https://gfx.nrk.no/cL1W3jptC9CW0XYwWVlrSgy4zzmodPTI5OtsbBHdtv1w.jpg

publishedAt     2021-06-18T21:43:59Z

content         BYD, NIO, Hongqi, Xpeng, Seres og Dongfeng. Kanskje ikke så kjente bilmerker for nordmenn flest, men snart kan de bli det.
For helkinesiske elbiler er på full fart inn i det norske markedet, bekreft… [+5142 chars]


In [63]:
#converting articles to data frame
df=pd.DataFrame(articles)
print(df.shape)
df

(20, 8)


source  \
0                        {'id': 'nrk', 'name': 'NRK'}   
1                 {'id': None, 'name': 'Mdig.com.br'}   
2                  {'id': None, 'name': 'Publico.pt'}   
3         {'id': None, 'name': 'Jornaldenegocios.pt'}   
4        {'id': 'independent', 'name': 'Independent'}   
5                    {'id': 'lenta', 'name': 'Lenta'}   
6        {'id': None, 'name': 'The Truth About Cars'}   
7            {'id': None, 'name': 'Donanimhaber.com'}   
8         {'id': None, 'name': 'Yahoo Entertainment'}   
9                    {'id': None, 'name': 'Barron's'}   
10                  {'id': None, 'name': 'CarScoops'}   
11             {'id': 'fox-news', 'name': 'Fox News'}   
12                   {'id': None, 'name': 'Electrek'}   
13  {'id': 'der-tagesspiegel', 'name': 'Der Tagess...   
14  {'id': 'der-tagesspiegel', 'name': 'Der Tagess...   
15                  {'id': None, 'name': 'PetaPixel'}   
16              {'id': None, 'name': 'Cointelegraph'}   
17                   {'id': None, 'name': 'Feber.se'}   
18        {'id': None, 'name': 'Yahoo Entertainment'}   
19        {'id': None, 'name': 'Yahoo Entertainment'}   

                                       author  \
0                         Lars Håkon Pedersen   
1           https://www.facebook.com/luisaocs   
2                    carla.ribeiro@publico.pt   
3                                      Àlvaro   
4                              Graig Graziosi   
5                               Марина Совина   
6                                  Tim Healey   
7                                 Barış Cankı   
8             Damanick Dantes and Frances Yue   
9                            Lauren R. Rublin   
10                             Sebastien Bell   
11                              Louis Casiano   
12                               Fred Lambert   
13                           Thorsten Metzner   
14                           Thorsten Metzner   
15                                David Crewe   
16  Cointelegraph By Cointelegraph Consulting   
17                                Roger Åberg   
18                               Debasis Saha   
19                              Asma UL Husna   

                                                title  \
0       Kinesisk sjarmoffensiv mot norske elbilkunder   
1      A antiga roda d'água revisitada como uma bomba   
2     Model 3, o Tesla que está cada vez mais europeu   
3            Forum Geral • Re: Impresa - Tópico Geral   
4   ‘Big Short’ investor Michael Burry is back on ...   
5   Ущерб от отмены фестиваля «Дикая мята» оценили...   
6          That Tesla Model S Plaid 0-60 Time is Bunk   
7   Milyarder Charles Hoskinson, stablecoin'ler ha...   
8   Market Wrap: Bitcoin Drops Ahead of Looming ‘D...   
9   Why Moderna, Illumina, or China’s Meituan Coul...   
10  30 Tesla Crashes With 10 Deaths Linked To Adva...   
11  California couple charged in alleged freeway s...   
12  Podcast: more info on Tesla Model S Plaid, yok...   
13  Erst Kräne, dann Pläne: Neuer 11 000-Seiten-An...   
14  Neuer 11.000-Seiten-Antrag für Fabrik in Grünh...   
15  This DIY Portable Power Station Can Keep Tons ...   
16  Cointelegraph Consulting: Report pictures a cr...   
17   Riktigt bra genomgång av nya Tesla Model S Plaid   
18                                Is JFU Stock A Buy?   
19                               Is PLPC Stock A Buy?   

                                          description  \
0   Elbilmarkedet i Norge har vært dominert av tys...   
1   A eletricidade é, junto com o fogo e a invençã...   
2   A estreia foi há pouco, mas já há versão aprim...   
3                                                       
4   Reddit investors largely shrugged off the inve...   
5   Организаторы фестиваля «Дикая мята» под Тулой,...   
6   Continuing a theme from earlier today, we need...   
7   IOHK CEO'su ve Cardano'nun yaratıcısı Charles ...   
8   Bitcoin’s ‘death cross’ could increase selling...   
9   Baillie Gifford’s Tom Slater invests in compan.

In [64]:
df.to_csv('result.csv')